syntax = "proto3";

// option optimize_for = CODE_SIZE;
// option cc_enable_arenas = true;

package bftrader.bfgateway;

import "bftrader.proto";

//
// 网关服务，必须先connect然后才可以调用其他api
// 本网关实现的是广播服务，用于做远程管理工具/远程策略引擎
// 不想做远程策略引擎的，可使用内置的Cta引擎
// 
service BfGatewayService{    
    // 请求建立到proxy的连接
    rpc Connect(BfConnectReq) returns (BfConnectResp) {}
    
    // 活跃检测
    rpc Ping(BfPingData) returns (BfPingData) {}
        
    // 请求断开到proxy的连接
    rpc Disconnect(BfVoid) returns (BfVoid) {}

    // 获取交易合约信息
    rpc GetContract(BfGetContractReq) returns (BfContractData) {}
       
    // 发单
    rpc SendOrder(BfSendOrderReq) returns (BfSendOrderResp) {}
    
    // 撤单
    rpc CancelOrder(BfCancelOrderReq) returns (BfVoid) {}
    
    // 查询账户信息
    rpc QueryAccount(BfVoid) returns (BfVoid) {}
    
    // 查询持仓信息
    rpc QueryPosition(BfVoid) returns (BfVoid) {}
}